controllers/queries.js

```diff
@@ -1,91 +1,85 @@
 var mongodb = require( 'mongodb' );
 var ObjectId = require( 'mongodb' ).ObjectID;
 var convertToMS = require( './convertToMS' );
+var Post = require( '../models/post.model' );
 
 
 module.exports = {
 
   // POST /add
 
-  postAdd: (req, res) => {
+  postAdd: ( req, res ) => {
     var username = req.body.username;
     var availableTime = req.body.availableTime;
     var setup = req.body.setup;
     var interests = req.body.interests;
-    var endTime = convertToMS(availableTime);
-    console.log( 'Received: ', username, availableTime, setup, interests);
-    var db = req.db;
-    var posts = db.collection( 'posts' );
-    posts.insert({
+    var endTime = convertToMS( availableTime );
+    if ( !username || !availableTime ) return res.status( 400 ).send( { status: 400, error: "Missing parameters." });
+    // create a new post
+    var newPost = Post({
       postTime: new Date().getTime(),
       username: username,
       endTime: new Date().getTime() + endTime,
       setup: setup,
       interests: interests
     });
-    res.redirect( 'http://pairboard.surge.sh' );
+    newPost.save( err => {
+      if ( err ) return handleError( err );
+      res.redirect( 'http://pairboard.surge.sh' );
+    })
   },
 
   // GET /posts
 
   getPosts: ( req, res ) => {
-    var db = req.db;
-    var posts = db.collection( 'posts' );
     var now = new Date().getTime();
-    posts.find({ endTime: { $gt: now }}).toArray( ( err, posts ) => {
-      if (err) {
-        console.error( err );
-      } else {
+    Post.find(
+      { endTime: { $gt: now }},
+      ( err, posts ) => {
+        if ( err ) return handleError( err );
         res.json( posts );
-      }
-    })
+      });
   },
 
   // POST /:id
 
-  deletePost: (req, res) => {
-    var db = req.db;
-    var id = new ObjectId(req.params.id);
-    var posts = db.collection( 'posts' );
-    try {
-      posts.deleteOne({ '_id': id });
-      console.log('Attempting to delete', id)
-    } catch( err ) {
-      console.log( err );
-    }
-    res.redirect( 'http://pairboard.surge.sh' );
+  deletePost: ( req, res ) => {
+    var id = new ObjectId( req.params.id );
+    if ( !ObjectID.isValid(id) ) return res.status( 400 ).send( {error: "Invalid id." })
```

Comment:
Right you are...I was thinking I was referencing the mongodb.ObjectID, not the variable I'd already made for it...